types/src/transaction/block_epilogue.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_crypto::HashValue;
+#[cfg(any(test, feature = "fuzzing"))]
+use proptest_derive::Arbitrary;
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+pub enum BlockEpiloguePayload {
+    BlockId(HashValue),
+    WithBlockEndInfo {
+        block_id: HashValue,
+        block_end_info: BlockEndInfo,
+    },
+}
+
+impl BlockEpiloguePayload {
+    pub fn try_as_block_end_info(&self) -> Option<&BlockEndInfo> {
+        match self {
+            BlockEpiloguePayload::BlockId(_) => None,
+            BlockEpiloguePayload::WithBlockEndInfo { block_end_info, .. } => Some(block_end_info),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct BlockEndInfo {
```

Comment:
Should this itself be an enum? I feel it's more likely to add things to "block end info" than to let block epilogue payload carry things other than "block end info".